### PR TITLE
[DOCS] Add template docs to scripts. Reorder template examples.

### LINKED ISSUE #42530: Docs: Clarify how to store a search template
https://www.elastic.co/guide/en/elasticsearch/reference/current/search-template.html
https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-scripting-using.html
https://www.elastic.co/guide/en/elasticsearch/client/java-api/current/java-search-template.html

Clarify in search template doc where it refers to storing a search template in a **Note** as a preregistered template (https://www.elastic.co/guide/en/elasticsearch/reference/current/search-template.html#pre-registered-templates). Clarify example for storing a search template.  Add link as appropriate from **How to use scripts** doc.

### PR DESCRIPTION
**Pull request by jrodewig**
* Adds a search template section to the [How to use scripts](https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-scripting-using.html) chapter. This section cross-links to the [Search template](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-template.html) chapter for detailed info and examples.

* Reorders and retitles several examples in the Search template chapter. This is primarily to make examples for storing, deleting, and using search templates more prominent. No substantive changes were made to the example sections.

Resolves #42530.

### Previews
- **How to use scripts**: http://elasticsearch_45817.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/modules-scripting-using.html#modules-scripting-search-templates
- **Search template**: http://elasticsearch_45817.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/search-template.html#_examples_34

### CODE REVIEW

File: docs/reference/search/search-template.asciidoc [hunk 1]
```diff
@@ -32,7 +32,209 @@ disable scripts per type and context as described in the
...
+
+[float]
+[[pre-registered-templates]]
+===== Store a search template
+
+You can store a search template using the stored scripts API.
+
+[source,js]
+------------------------------------------
+POST _scripts/<templatename>
```
Comment (rjernst) [inline docs/reference/search/search-template.asciidoc]: I believe we've used the terminology "id" both in code and api for stored scripts. We should stay consistent here and call this `templateid`?

**Review by rjernst — APPROVED** [review]
LGTM, one suggestion

File: docs/reference/search/search-template.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/search/search-template.asciidoc]: Good call! Fixed with 71d03edd885.

**Comment by untergeek** [comment]
Thank you! This is wonderful!

**Pull request merged**